src/content/tools/hooks.md

```diff
@@ -0,0 +1,294 @@
+
+### CodeAsset type {:.no_toc}
+
+A [`CodeAsset`][] represents a code asset. A code asset is
+a dynamic library compiled from a language other than
+Dart, such as C, C++, Rust, or Go. `CodeAsset` is part of
+the `code_asset` package and are called with `dart:ffi` from
+the Dart code at runtime using the `@Native` annotation.
+
+[`CodeAsset`]: {{site.pub-api}}/code_assets/latest/code_assets/CodeAsset-class.html
+
+## Use a hook {: #use-hooks-assets }
+
+To add assets to your project, use a hook. For details,
+see the following sections.
+
+### Add dependencies {: #add-dependencies-hooks-assets }
+
+To use a hook, you must first add it to your `pubspec.yaml`
+dependencies.
+
+1. Open your package in a code editor.
+
+1. Navigate to `pubspec.yaml`.
+
+1. Go to the `dependencies` section and add the following:
+
+    * `hooks`
+    * `code_assets`
+
+1. To this list, also add any tools for your specific task,
+   such as `native_toolchain_c` for compiling C code.
+
+:::note
+You need to add the dependencies under `dependencies`, not
+`dev_dependencies`. The hooks will be run by packages and
+applications depending on your package, so the
+Dart code needs to be part of the resolution in those
+packages.
+:::
+
+Example dependencies for a build hook:
+
+```yaml title="pubspec.yaml"
+name: native_add_library
+description: Sums two numbers with native code.
+version: 0.1.0
+
+environment:
+  sdk: '>=3.9.0 <4.0.0'
+
+dependencies:
+  ... 
+  code_assets: any
+  hooks: any
+  native_toolchain_c: any
+
+dev_dependencies:
+  ...
+  ffigen: ^18.0.0
+ ```
+
+### Create a build hook to generate native assets {: #create-hook }
+
+If you want to use a build hook to transparently compile
+native assets (such as C or Rust libraries), which are then
+made available to be called from the Dart code of a package,
+create a `build.dart` script similar to the following:
+
+1.  In your Dart project, create or open `hooks/build.dart`. 
+
+1.  In the `main` method, add a `build` function from
+    `package:hooks/hooks.dart` and use the appropriate
+    toolchain to compile the native library. For example:
+
+    ```dart title="hooks/build.dart" highlightLines=6
+    import 'package:hooks/hooks.dart';
+    import 'package:logging/logging.dart';
+    import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+    void main(List<String> args) async {
+      await build(args, (input, output) async {
+        final packageName = input.packageName;
+        final cbuilder = CBuilder.library(
+          name: packageName,
+          assetName: '$packageName.dart',
+          sources: ['src/$packageName.c'],
+        );
+        await cbuilder.run(
+          input: input,
+          output: output,
+          logger: Logger('')
+            ..level = Level.ALL
+            ..onRecord.listen((record) => print(record.message)),
+        );
+      });
+    }
+    ```
+
+    The `build` function takes two arguments:
+
+    * `input`: (readonly) The input for the hook.
```

Comment:
We don't use this `(readonly)` syntax anywhere else, so consider including that information directly in the sentence.

```suggestion
    * `input`: The read-only input for the hook.
```